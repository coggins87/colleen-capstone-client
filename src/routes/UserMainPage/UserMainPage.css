.user_main {
  padding-top: 13%;
  padding-left: 5%;
  text-align: center;
  background-color: rgb(17, 0, 4);

}

.user_header {
  margin-bottom: 1%;
  color: rgb(2, 69, 114);
}

.user_content {
  padding-bottom: 10%;
  display: flex;
  flex-direction: column;
}

@media all and (max-width: 500px) {
  .user_main {
    padding-top: 35%;
  }
}
@media all and (max-width: 350px) {
  .user_main {
    padding-top: 40%;

    margin-bottom: 5%;
  }
}
