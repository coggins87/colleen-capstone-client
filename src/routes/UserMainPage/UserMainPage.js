import React from "react";
import WorkoutList from "../../components/WorkoutList/workout-list";
import WorkoutSearchService from "../../services/workout-search-service";
import "./UserMainPage.css";
class UserMainPage extends React.Component {
  state = {
    workouts: [],
    error: null
  };
  static defaultProps = {
    match: { params: {} }
  };

  componentDidMount() {
    const userId = this.props.match.params.userId;

    WorkoutSearchService.getUserWorkouts(userId)
      .then(res => {
        this.setState({
          workouts: res
        });
      })
      .catch(res => {
        if (res.message) {
          this.setState({ error: "Something went wrong! Try again later." });
        } else {
          this.setState({ error: res.error });
        }
      });
  }
  render() {
    const error = this.state.error;
    return (
      <div className="user_main">
        <h2>My Saved Workouts</h2>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="user_content">
          {this.state.workouts.length === 0 ? (
            <p>
              Search for a workout, if you like what you find, save it here for
              later!
            </p>
          ) : (
            <WorkoutList workouts={this.state.workouts} />
          )}
        </div>
      </div>
    );
  }
}

export default UserMainPage;
