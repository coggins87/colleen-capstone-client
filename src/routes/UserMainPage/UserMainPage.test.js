import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import UserMainPage from './UserMainPage'


describe('UserMainPage component', ()=>{
  const props =  {
    match: { params: {} }
  };

  it('renders the UserMainPage component by default', () => {
    const wrapper = shallow(<UserMainPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders the UserMainPage component given props', () => {
    const wrapper = shallow(<UserMainPage {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })


})

