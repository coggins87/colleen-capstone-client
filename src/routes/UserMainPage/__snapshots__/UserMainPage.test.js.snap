// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMainPage component renders the UserMainPage component by default 1`] = `
<div
  className="user_main"
>
  <h2>
    My Saved Workouts
  </h2>
  <div
    role="alert"
  />
  <div
    className="user_content"
  >
    <p>
      Search for a workout, if you like what you find, save it here for later!
    </p>
  </div>
</div>
`;

exports[`UserMainPage component renders the UserMainPage component given props 1`] = `
<div
  className="user_main"
>
  <h2>
    My Saved Workouts
  </h2>
  <div
    role="alert"
  />
  <div
    className="user_content"
  >
    <p>
      Search for a workout, if you like what you find, save it here for later!
    </p>
  </div>
</div>
`;
